dist/svg.js

```diff
@@ -1409,23 +1415,14 @@ SVG.FX = SVG.invent({
       this.situation = this.situations.shift()
 
       if(this.situation){
-
-        var fn = function(){
-          if(this.situation instanceof SVG.Situation)
-            this.initAnimations().atStart()
-          else if(this.situation instanceof SVG.Delay)
-            this.dequeue()
-          else
-            this.situation.call(this)
-        }.bind(this)
-
-        // start next animation
-        if(this.situation.delay){
-          setTimeout(function(){fn()}, this.situation.delay)
-        }else{
-          fn()
+        if(this.situation instanceof SVG.Situation) {
+          this.startCurrent()
+        } else {
+          // If it is not a SVG.Situation, then it is a function, we execute it
+          // and then pull the next element from the queue
+          this.situation.call(this)
+          return this.dequeue()
```

Comment:
there is a good reason not to have `dequeue` called here. The reason is simply, that `dequeue` should be called **inside** the function. So this function can do asyncronous operations (like getting data via ajax or just animating some other stuff) and call `dequeue` when its done.
So this was intended and should be done imo since you loose any control as a user when dequeue is called immeditaley